### PR TITLE
PFM-515

### PR DESCRIPTION
**Pull request by jyotishk-eGov**
Search LOI changes

### CODE REVIEW

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            String tenantId = rs.getString("tenantId");
+            String loiNumber = rs.getString("loi_number");
+            String workPkgNumber = rs.getString("work_pkg_number");
+            String workIdentificationNumber = rs.getString("work_identification_number");
+            String fileNumber = rs.getString("file_number");
+            BigDecimal negotiatedPercentage = rs.getBigDecimal("negotiated_percentage");
+            BigDecimal agreementDate = rs.getBigDecimal("agreement_date");
+            String contractorId = rs.getString("contractor_id");
+            BigDecimal securityDeposit = rs.getBigDecimal("security_deposit");
+            String bankGaurante = rs.getString("bank_gaurante");
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java:38]: I have renamed the attribute in the db as the first one had a typo.

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            BigDecimal emdAmount = rs.getBigDecimal("emd_amount");
+            BigDecimal contractPeriod = rs.getBigDecimal("contract_period");
+            BigDecimal defectLiabilityPeriod = rs.getBigDecimal("defect_liability_period");
+            String oicId = rs.getString("oic_id");
+            String status = rs.getString("status");
+            String letterStatus = rs.getString("letter_status");
+            String additionalDetails = rs.getString("additionaldetails");
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java:45]: This is jaon. Not a string.

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            Long createdtime = rs.getLong("createdtime");
+            Long lastmodifiedtime = rs.getLong("lastmodifiedtime");
+
+
+            AuditDetails auditDetails = AuditDetails.builder().createdBy(createdby).createdTime(createdtime)
+                    .lastModifiedBy(lastmodifiedby).lastModifiedTime(lastmodifiedtime)
+                    .build();
+
+
+            LetterOfIndent estimate = LetterOfIndent.builder().id(UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8)))
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java:57]: Why have you named the vatiable estimate?

File: backend/loi-service/src/main/java/org/egov/works/web/models/LOISearchRequest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import org.egov.common.contract.request.RequestInfo;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+
+
+public class LOISearchRequest {
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/web/models/LOISearchRequest.java:14]: Do we need this?

File: backend/loi-service/src/main/java/org/egov/works/web/models/LetterOfIndent.java [hunk 5]
```diff
@@ -71,15 +71,22 @@ public class LetterOfIndent {
...
     private String letterStatus = null;
+
     @JsonProperty("auditDetails")
     private AuditDetails auditDetails = null;
+
     @JsonProperty("additionalDetails")
     private Object additionalDetails = null;
 
+    @JsonProperty("bankGaurante")
+    private Object bankGaurante = null;
```
Comment (rushang7-eGov) [inline backend/loi-service/src/main/java/org/egov/works/web/models/LetterOfIndent.java:88]: Check the updated name from the loi-service branch. 
I think the type is String. 
You will get all the changes when merge those changes with your code.

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]
Remove the .class file from git. 
Merge the loi-service branch and resolve all the conflicts.

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java (hunk 1 above)
Comment (jyotishk-eGov) [reply backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java]: Fixed.

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java (hunk 2 above)
Comment (jyotishk-eGov) [reply backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java]: Fixed.

File: backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java (hunk 3 above)
Comment (jyotishk-eGov) [reply backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java]: renamed to letterOfIntent

**Review by rushang7-eGov — COMMENTED** [review]
Resolve the merge conflicts

**Comment by jyotishk-eGov** [comment]
No need to merge these changes

**Pull request closed without merging**